Mathlib/NumberTheory/EulerProduct/Basic.lean

```diff
@@ -60,36 +67,87 @@ where `R` is a complete normed commutative ring. The main result is `EulerProduc
 variable {R : Type*} [NormedCommRing R] [CompleteSpace R] {f : ℕ → R}
 variable (hf₁ : f 1 = 1) (hmul : ∀ {m n}, Nat.Coprime m n → f (m * n) = f m * f n)
 
-/-- We relate a finite product over primes to an infinite sum over smooth numbers. -/
-lemma summable_and_hasSum_smoothNumbers_prod_primesBelow_tsum
-    (hsum : ∀ {p : ℕ}, p.Prime → Summable (fun n : ℕ ↦ ‖f (p ^ n)‖)) (N : ℕ) :
-    Summable (fun m : N.smoothNumbers ↦ ‖f m‖) ∧
-      HasSum (fun m : N.smoothNumbers ↦ f m) (∏ p in N.primesBelow, ∑' (n : ℕ), f (p ^ n)) := by
-  induction' N with N ih
-  · rw [smoothNumbers_zero, primesBelow_zero, Finset.prod_empty]
+-- local instances to speed up typeclass search
+@[nolint defLemma docBlame] local instance : TopologicalSpace R := UniformSpace.toTopologicalSpace
+@[nolint defLemma docBlame] local instance : T0Space R := MetricSpace.instT0Space
+@[nolint defLemma docBlame] local instance : RegularSpace R := TopologicalAddGroup.regularSpace R
+@[nolint defLemma docBlame] local instance : TopologicalSpace.MetrizableSpace R :=
+  MetricSpace.toMetrizableSpace
+@[nolint defLemma docBlame] local instance : T2Space R :=
+  TopologicalSpace.t2Space_of_metrizableSpace
```

Comment:
I'll check this again.

Reply:
OK, there is really not that much difference. I left in one of these (which gives a speed-up of ~50ms). (Regarding the linters complaining about local instances, see [here](https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/defLemma.20and.20docBlame.20linters.20on.20local.20instances.20in.20sections/near/433824150) on Zulip.)

What happened was this. I worked on a copy of this file in a separate project to try out things, and I had two more imports there (because further below I had a copy of the contents of another file that needed them). These additional imports led to a 2x slow-down of typeclass search. When I copied my new version over to Mathlib, I left out the two imports...

(But this shows that these slow typeclass instance search problems are very sensitive to context, which is a bit problematic.)